file_util/inc/leatherman/file_util/directory.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file
+ * Declares utility functions for enumerating directories.
+ */
+#pragma once
+
+#include <string>
+#include <functional>
+
+namespace leatherman { namespace file_util {
```

Comment:
Do we want to preserve distinct namespaces for file and directory? There aren't any conflicts, so seems safe enough to put them all in one namespace.